server/game_engine/GameEngine.js

```diff
@@ -144,30 +145,74 @@ class Game {
             "Car", "Airplane", "Bike", "Truck", "Tesla",
             "Sword", "Shield", "Staff", "Bow", "Helmet", "Dagger", "Gun", "Belt", "Armor",
             "Beach", "Jungle", "Desert", "Water", "Earth", "Wind", "Fire", "Grass", "Space", "Snow", "Moon", "Electricity",
             "Ball", "Tail", "Shoe", "Rainbow", "Pole", "Computer", "Cellphone", "Camera", "Bitcoin", "Money", "Book",
             "Television", "House", "Doll",
             "Run", "Change", "Teleport", "Slash", "Switch", "Eat", "Picture", "Dare", "Retire"];
-        var colorsIndex = [9, 8, 7, 1]; //Counter for card colors to be distributed (9 blue, 8 red, 7 white, 1 black)
-        var colors = ["blue", "red", "white", "black"]
+        let colorsIndex = [9, 8, 7, 1]; //Counter for card colors to be distributed (9 blue, 8 red, 7 white, 1 black)
+        let colors = ["blue", "red", "white", "black"]
         this.board = []; //Reset the board
-        for (var i = 0; i < 5; i++) {
+        for (let i = 0; i < 5; i++) {
             this.board.push([]); //Push empty row
-            for (var j = 0; j < 5; j++) {
+            for (let j = 0; j < 5; j++) {
                 const randomWordIndex = Math.floor(Math.random() * words.length); //Pick random word
                 const randomColorIndex = Math.floor(Math.random() * colors.length); //Pick random color
-                var card = { word: words[randomWordIndex], color: colors[randomColorIndex], revealed: false }
+                let card = { word: words[randomWordIndex], color: colors[randomColorIndex], revealed: false }
                 words.splice(randomWordIndex, 1); //Remove the word from the list so it can't be picked again.
                 // If the limit for a specific color is reached, remove it from the list, otherwise decrement it.
                 colorsIndex[randomColorIndex] === 1 ? (colors.splice(randomColorIndex, 1), colorsIndex.splice(randomColorIndex, 1)) : colorsIndex[randomColorIndex]--;
                 this.board[i].push(card);
             }
         }
     }
+
+    getRoles() {
+        return {
+            blueGuessers: this.blueGuessers,
+            blueSpymaster: this.blueSpymaster,
+            redGuessers: this.redGuessers,
+            redSpymaster: this.redSpymaster
+        }
+    }
+
+    assignRole(playerId, role) {
+        switch (role) {
+            case "blueSpymaster":
+                this.blueSpymaster = playerId;
+                break;
+            case "redSpymaster":
+                this.redSpymaster = playerId;
+                break;
+            case "blueGuesser":
+                this.blueGuessers.push(playerId);
+                break;
+            case "redGuesser":
+                this.redGuessers.push(playerId);
+                break;
+            default:
+                console.log("Error, something went wrong.")
+        }
+    }
+
+    to_json() {
```

Comment:
```suggestion
    toJson() {
```
To match JS convention